apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/notice.js

```diff
@@ -1,98 +1,124 @@
-					'full-site-editing'
-				),
-				{
-					a: (
-						<ExternalLink
-							href={ wpcomGlobalStyles.upgradeUrl }
-							target="_blank"
-							onClick={ recordUpgradePreSaveNoticeClick }
-						/>
-					),
-				}
-			) }
-		</Notice>
+	const { createWarningNotice, removeNotice } = useDispatch( 'core/notices' );
+	const { editEntityRecord } = useDispatch( 'core' );
+
+	const trackEvent = useCallback(
+		( eventName ) =>
+			recordTracksEvent( eventName, {
+				context: isSiteEditor ? 'site-editor' : 'post-editor',
+				blog_id: wpcomGlobalStyles.wpcomBlogId,
+			} ),
+		[ isSiteEditor ]
 	);
-}
 
-export default function GlobalStylesNotice() {
-	const { globalStylesConfig, isSaveViewOpened } = useSelect( ( select ) => ( {
-		globalStylesConfig: select( 'core' ).getEntityConfig( 'root', 'globalStyles' ),
-		isSaveViewOpened: select( 'core/edit-site' ).isSaveViewOpened(),
-	} ) );
+	const upgradePlan = useCallback( () => {
+		window.open( wpcomGlobalStyles.upgradeUrl, '_blank' ).focus();
+		trackEvent( 'calypso_global_styles_gating_notice_upgrade_click' );
+	}, [ trackEvent ] );
 
-	const [ isNoticeRendered, setIsNoticeRendered ] = useState( false );
+	const previewSite = useCallback( () => {
+		window.open( wpcomGlobalStyles.previewUrl, '_blank' ).focus();
+		trackEvent( 'calypso_global_styles_gating_notice_preview_click' );
+	}, [ trackEvent ] );
 
-	useEffect( () => {
-		if ( ! globalStylesConfig || ! isSaveViewOpened ) {
-			setIsNoticeRendered( false );
+	const resetGlobalStyles = useCallback( () => {
+		if ( ! globalStylesId ) {
 			return;
 		}
-		if ( isNoticeRendered ) {
-			return;
+
+		editEntityRecord( 'root', 'globalStyles', globalStylesId, {
+			styles: {},
+			settings: {},
+		} );
+
+		trackEvent( 'calypso_global_styles_gating_notice_reset_click' );
+	}, [ editEntityRecord, globalStylesId, trackEvent ] );
+
+	const showNotice = useCallback( () => {
+		const actions = [
+			{
+				label: __( 'Upgrade now', 'full-site-editing' ),
+				onClick: upgradePlan,
+				variant: 'primary',
+				noDefaultClasses: true,
+			},
+		];
+
+		if ( isPostEditor ) {
+			actions.push( {
+				label: __( 'Preview your site', 'full-site-editing' ),
+				onClick: previewSite,
+				variant: 'secondary',
+				noDefaultClasses: true,
+			} );
 		}
 
-		const preSavePanelTitles = document.querySelectorAll(
-			'.entities-saved-states__panel .components-panel__body.is-opened .components-panel__body-title'
-		);
+		if ( isSiteEditor ) {
+			actions.push( {
+				label: __( 'Remove custom styles', 'full-site-editing' ),
+				onClick: resetGlobalStyles,
+				variant: 'link',
+				noDefaultClasses: true,
+			} );
+		}
 
-		for ( const entityTitle of preSavePanelTitles ) {
-			if ( entityTitle.textContent !== globalStylesConfig.label ) {
-				continue;
+		createWarningNotice(
+			__(
+				'Your site includes customized styles that are only visible to visitors after upgrading to the Premium plan or higher.',
```

Comment:
What if we add a link on the `customized styles` that:
- In Site Editor will open the Global Styles sidebar 
- In the Post Editor will be a link to Site Editor, that would open the GS sidebar automatically (optional).

I think this would be helpful for users that come from onboarding and are unfamiliar with Global Styles (since they only interacted with Styles variations, fonts or colors).

I don't think we should point to the docs because:
 - We already do that on the front-end
 - They are already in edit mode; there's no point in redirecting them to docs and then coming back again to the editor to try it out.

Reply:
Unfortunately, we cannot add links without using the experimental `__unstableHTML` property which is highly discouraged: https://github.com/WordPress/gutenberg/blob/6606da447be0ba0885573add567dbb97b42c8ed2/packages/notices/src/store/selectors.js#L29-L33